sno/branch.py

```diff
@@ -0,0 +1,86 @@
+import json
+import sys
+
+import click
+import pygit2
+
+from .cli_util import MutexOption
+from .exec import execvp
+
+
+@click.command(context_settings=dict(ignore_unknown_options=True))
+@click.pass_context
+@click.option(
+    "--text",
+    "is_output_json",
+    flag_value=False,
+    default=True,
+    help="Get the status in text format",
+    cls=MutexOption,
+    exclusive_with=["json"],
+)
+@click.option(
+    "--json",
+    "is_output_json",
+    flag_value=True,
+    help="Get the status in JSON format",
+    cls=MutexOption,
+    exclusive_with=["text"],
+)
+@click.argument("args", nargs=-1, type=click.UNPROCESSED)
+def branch(ctx, is_output_json, args):
+    """ List, create, or delete branches """
+    repo_path = ctx.obj.repo_path
+    repo = ctx.obj.repo
+
+    sargs = set(args)
+    if is_output_json:
+        if sargs - {"--list"}:  # sno branch --list --json is allowed.
```

Comment:
```suggestion
        if "--list" not in sargs:  # sno branch --list --json is allowed.
```

Reply:
This has a different behavior. Here is what is allowed:
git branch --json
git branch --list --json 

Here is what is not allowed:
git branch -D delete_me --json

So I'm testing if sargs is empty, or just contains list. I'll rearrange to make it a bit clearer.